In [1]:
import numpy as np
import pandas as pd
import scipy as sp
from scipy.optimize import minimize

## Problema principal

Buscamos minimizar las cantidades 
$$K_1=(b+c-a)C_1$$ y $$K_2=(b+c-a)C_2$$ provenientes de la función
$$K(\tau)=K_1+K_2\tau^{1/2}-\tfrac{2}{3\sqrt{3}}(\tfrac{\tau}{2}-1)^{3/2}.$$

Para esto, establecemos el problema de minimización $(P)$ dado por
$$\min_{(a, b, c, C_1, C_2)\in X}\ K_1^2+K_2^2$$
donde $X\subset \mathbb{R}^5$ es el espacio de restricciones del problema.

## Problema de minimización

Establecemos la función objetivo y las restricciones del problema $(P)$.

*_Observación_*: por comodidad, en el código se consideró $C_1=A^{-1}$ y $C_2=B^{-1}$.

In [2]:
def objective(x):
    a=x[0]
    b=x[1]
    c=x[2]
    A=x[3]
    B=x[4]
    return ((b+c-a)**2)*(1/(A**2)+1/(B**2))
def constraint1(x):
    a=x[0]
    b=x[1]
    c=x[2]
    A=x[3]
    B=x[4]
    return 14*a*(b-2*a)**4-1.5*(b**2)*(A**2)-1.5*(b+c-a)*a*b*(A**2)-1.5*a*(b**2)*(B**2)-0.75*(b+c-a)*(A**4)-0.125*b*(A**2)*(B**2)
def constraint2(x):
    a=x[0]
    b=x[1]
    c=x[2]
    A=x[3]
    B=x[4]
    return 6*a*(b-2*a)**2-1.5*b*(A**2)-0.75*a*(B**2)
def constraint3(x):
    return x[1]-2*x[0]

In [3]:
x0=[1, 5, 1, 3, 3]

#restricciones
con1={'type':'ineq', 'fun': constraint1}
con2={'type':'ineq', 'fun': constraint2}
con3={'type':'ineq', 'fun': constraint3}
cons=[con1, con2, con3]

## Resolución

In [4]:
lista=np.linspace(1, 100, 300).tolist()
val_p_min=100
sol_p_min=[10, 10, 10, 10, 10]

for j in lista:
    b1=(1, j)
    b=(0.01, None)
    # Cotas de las variables
    bnds=(b, b, b, b1, b1)
    sol=minimize(objective, x0, bounds=bnds, constraints=cons)
    # Nos aseguramos de que, en efecto, se satisfagan las restricciones
    # Guardamos la mínima solución
    if sol.fun<val_p_min and constraint1(sol.x)>0 and constraint2(sol.x)>0 and constraint3(sol.x)>0:
        val_p_min=sol.fun
        sol_p_min=sol.x        

print("---------------------------- Solución ----------------------------")
print("val(P)=", val_p_min)
print("Solución del problema: a=%s, b=%s, c=%s, C1=1/%s, C2=1/%s" %tuple(sol_p_min))
print("El valor de K1 es ", (sol_p_min[1]+sol_p_min[2]-sol_p_min[0])*(1/sol_p_min[3]))
print("El valor de K2 es ", (sol_p_min[1]+sol_p_min[2]-sol_p_min[0])*(1/sol_p_min[4]))

---------------------------- Solución ----------------------------
val(P)= 3.844302926319496
Solución del problema: a=22.489154744783647, b=117.75151503430727, c=0.01, C1=1/55.59593000565593, C2=1/99.99999999999676
El valor de K1 es  1.713657101874747
El valor de K2 es  0.9527236028952673


### Conclusión

Para $C_1, C_2\geq 1/100$ consideramos la solución aproximada a tres decimales

$$a=22.489, b=117.752, c=0.01, C_1=1/55.595, C_2=1/100,$$
donde
$$val(P_{aprox})=3.84445,$$
que es mayor a $val(P)$ y difere de este en $1.5\cdot 10^{-4}$. 

A partir de esta elección se tiene que $\eta\geq 13.338.$

Las conclusiones son similares al problema $1$.